common/utilities/time/waiting-on.h

```diff
@@ -87,16 +87,17 @@ class waiting_on
         std::shared_ptr< std::nullptr_t > const _invalidator;
 
     public:
-        in_thread_( waiting_on const& local )
+        in_thread_( waiting_on const & local )
             : _ptr( local._ptr )
-            , _invalidator(nullptr, [ weak_ptr = std::weak_ptr< wait_state_t >(local._ptr)]( std::nullptr_t * )
-                {
-                // We get here when the lambda we're in is destroyed -- so either we've already run
-                // (and signalled once) or we've never run. We signal anyway -- if anything's
-                // waiting they'll get woken up; otherwise nothing'll happen...
-                if( auto wait_state = weak_ptr.lock() )
-                    wait_state->invalidate();
-                })
+            , _invalidator(
+                  nullptr,
+                  [weak_ptr = std::weak_ptr< wait_state_t >( local._ptr )]( std::nullptr_t * ) {

```

Comment:
I think we need the lambda to get `std::nullptr_t` as an argument, no?

Reply:
No, libCi had errors on it saying it should be std::nullptr_t *,
Now it passed